### PR TITLE
refactor(users): Move hardcoded email subjects to constants

### LINKED ISSUE #5940: [REFACTOR] Move Hardcoded Email Subjects to Constants Across the User Related Files
### Feature Description

In the current implementation, email subjects are hardcoded at multiple places across the codebase when constructing email content. To improve maintainability and avoid magic strings, each unique subject should be moved to its own constant. The goal of this issue is to identify all the places where email subjects are hardcoded, create a corresponding constant for each one, and replace the hardcoded values with the newly defined constants.

### Possible Implementation

- Search the codebase for all instances where email subjects are hardcoded.
  Example: https://github.com/juspay/hyperswitch/blob/b2364414edab6d44e276f2133bc988610693f155/crates/router/src/core/user.rs#L707
- For each unique subject, create a new constant in `crates/router/src/consts/user.rs`.
- Refactor the code by replacing the hardcoded subjects with their respective constants.

### Have you spent some time checking if this feature request has been raised before?

- [X] I checked and didn't find a similar issue

### Submission Process:
- Ask the maintainers for assignment of the issue, you can request for assignment by commenting on the issue itself.
- Once assigned, submit a pull request (PR).
- Maintainers will review and provide feedback, if any. 
- Maintainers can unassign issues due to inactivity, [read more here](https://github.com/juspay/hyperswitch/wiki/Hacktoberfest-Contribution-Rules).

Refer [here](https://github.com/juspay/hyperswitch/blob/main/docs/TERMS_OF_CONTEST.md) for Terms and conditions for the contest.

### PR DESCRIPTION
**Pull request by SummerGram**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Change the hardcoded email subject to constants

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
refactor based on this issue -  #5940 
[link](https://github.com/juspay/hyperswitch/issues/5940)
Closes #5940.

## How did you test it?

This is an internal change for email subjects.

There should be no affect. All email subjects like magic link, invite should be properly sent as before.

[change_apply_before.txt](https://github.com/user-attachments/files/17147768/change_apply_before.txt)
[change_apply_after.txt](https://github.com/user-attachments/files/17147771/change_apply_after.txt)

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by ThisIsMani** [comment]
Fix the PR title and link the issue. I'll approve the workflow.

**Comment by SummerGram** [comment]
What is the title of the PR supposed be?

**Comment by SummerGram** [comment]
@ThisIsMani 

I link the issue in the motivation and context part but I still cannot pass the check.

File: crates/router/src/core/recon.rs [hunk 1]
```diff
@@ -39,7 +39,8 @@ pub async fn send_recon_request(
         .attach_printable("Failed to convert recipient's email to UserEmail")?,
         settings: state.conf.clone(),
         subject: format!(
-            "Dashboard Pro Feature Request by {}",
+            "{} {}",
+            consts::user::EMAIL_SUBJECT_DASHBOARD_REQUEST,
```
Comment (ThisIsMani) [inline crates/router/src/core/recon.rs:43]: Can we move the constants related to Recon files in to this file instead
https://github.com/juspay/hyperswitch/blob/5912936f9f2178f002c96e25636b25f532b1ecb2/crates/router/src/consts.rs#L137.

File: crates/router/src/core/recon.rs [hunk 2]
```diff
@@ -145,7 +146,7 @@ pub async fn recon_merchant_account_update(
             .change_context(errors::ApiErrorResponse::InternalServerError)
             .attach_printable("Failed to form username")?,
         settings: state.conf.clone(),
-        subject: "Approval of Recon Request - Access Granted to Recon Dashboard",
+        subject: consts::user::EMAIL_SUBJECT_APPROVAL_RECON_REQUEST,
```
Comment (ThisIsMani) [inline crates/router/src/core/recon.rs:149]: Same here as well.

File: crates/router/src/consts/user.rs [hunk 3]
```diff
@@ -26,3 +26,14 @@ pub const REDIS_TOTP_SECRET_TTL_IN_SECS: i64 = 15 * 60; // 15 minutes
 
 pub const REDIS_SSO_PREFIX: &str = "SSO_";
 pub const REDIS_SSO_TTL: i64 = 5 * 60; // 5 minutes
+
+/// Email subject
+pub const EMAIL_SUBJECT_WELCOME: &str = "Welcome to the Hyperswitch community!";
```
Comment (ThisIsMani) [inline crates/router/src/consts/user.rs:31]:
```suggestion
pub const EMAIL_SUBJECT_SIGNUP: &str = "Welcome to the Hyperswitch community!";
```

File: crates/router/src/consts/user.rs [hunk 4]
```diff
@@ -26,3 +26,14 @@ pub const REDIS_TOTP_SECRET_TTL_IN_SECS: i64 = 15 * 60; // 15 minutes
...
+pub const EMAIL_SUBJECT_INVITATION: &str = "You have been invited to join Hyperswitch Community!";
+pub const EMAIL_SUBJECT_UNLOCK: &str = "Unlock Hyperswitch: Use Your Magic Link to Sign In";
```
Comment (ThisIsMani) [inline crates/router/src/consts/user.rs:33]:
```suggestion
pub const EMAIL_SUBJECT_MAGIC_LINK: &str = "Unlock Hyperswitch: Use Your Magic Link to Sign In";
```

File: crates/router/src/consts/user.rs [hunk 5]
```diff
@@ -26,3 +26,14 @@ pub const REDIS_TOTP_SECRET_TTL_IN_SECS: i64 = 15 * 60; // 15 minutes
...
+pub const EMAIL_SUBJECT_RESET_PASSWORD: &str = "Get back to Hyperswitch - Reset Your Password Now";
+pub const EMAIL_SUBJECT_API_KEY_EXPIRY: &str = "API Key Expiry Notice";
```
Comment (ThisIsMani) [inline crates/router/src/consts/user.rs:35]: Move this as well to `consts.rs`.

File: crates/router/src/consts/user.rs [hunk 6]
```diff
@@ -26,3 +26,14 @@ pub const REDIS_TOTP_SECRET_TTL_IN_SECS: i64 = 15 * 60; // 15 minutes
...
+pub const EMAIL_SUBJECT_DASHBOARD_FEATURE_REQUEST: &str = "Dashboard Pro Feature Request by";
+pub const EMAIL_SUBJECT_APPROVAL_RECON_REQUEST: &str =
+    "Approval of Recon Request - Access Granted to Recon Dashboard";
+pub const EMAIL_SUBJECT_NEW_PROD: &str = "New Prod Intent";
```
Comment (ThisIsMani) [inline crates/router/src/consts/user.rs:39]:
```suggestion
pub const EMAIL_SUBJECT_NEW_PROD_INTENT: &str = "New Prod Intent";
```

**Review by ThisIsMani — CHANGES_REQUESTED** [review]

**Comment by SummerGram** [comment]
@ThisIsMani 
Please review the change.

**Review by ThisIsMani — APPROVED** [review]

**Comment by SummerGram** [comment]
@ThisIsMani 

Why is it failed to run?

File: crates/router/src/core/recon.rs [hunk 7]
```diff
@@ -39,7 +39,8 @@ pub async fn send_recon_request(
         .attach_printable("Failed to convert recipient's email to UserEmail")?,
         settings: state.conf.clone(),
         subject: format!(
-            "Dashboard Pro Feature Request by {}",
+            "{} {}",
+            consts::EMAIL_SUBJECT_DASHBOARD_REQUEST,
```
Comment (ThisIsMani) [inline crates/router/src/core/recon.rs:43]:
```suggestion
            consts::EMAIL_SUBJECT_DASHBOARD_FEATURE_REQUEST,
```

**Comment by ThisIsMani** [comment]
Please recheck the variables again.

**Comment by SummerGram** [comment]
> Please recheck the variables again.

Those failure in ci-pr is related to the variables name?

**Comment by ThisIsMani** [comment]
You code is not compiling. That's why the run is failed.

You can check by running `just check`.

**Comment by SummerGram** [comment]
> You code is not compiling. That's why the run is failed.
> 
> You can check by running `just check`.

Thanks!

**Comment by SanchithHegde** [comment]
> I think "clonable" typo is not related to this PR.

You can ignore that, shouldn't be a problem. Moreover, it's not a required check, so it shouldn't block your PR from being merged.

**Review by ThisIsMani — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**